src/gui/src/heatMap.cpp

```diff
@@ -850,31 +850,31 @@ void RealValueHeatMapDataSource::correctMapScale(HeatMapDataSource::Map& map)
 {
   determineMinMax(map);
   determineUnits();
-  min_ = roundData(min_);
-  max_ = roundData(max_);
 
   for (const auto& map_col : map) {
     for (const auto& map_pt : map_col) {
       map_pt->value = convertValueToPercent(map_pt->value);
     }
   }
 
+  min_ = roundData(min_ * scale_);
+  max_ = roundData(max_ * scale_);
+
   // reset since all data has been scaled by the appropriate amount
   scale_ = 1.0;
 }
 
 double RealValueHeatMapDataSource::roundData(double value) const
 {
   const double precision = 1000.0;
-  double new_value = value * scale_;
-  return std::round(new_value * precision) / precision;
+  return std::round(value * precision) / precision;
 }
```

Comment:
Does this change the precision now that you are not multiplying by scale_?

Reply:
I moved the min and max values to be scaled after all the values are processed. So the scaling should not be needed since all the values have the same scale. (Doing the scaling across two functions was also somewhat confusing when debugging).

Reply:
Sure but does it affect the final rounded value?  If scale_ != 1.0 then this will result in a different return value.

Reply:
it should not, there were only two places where the `scale_` was used (for the min and max rounding) and I just make sure it's multiplied in the argument in the function call, effectively should be the same answer.

Reply:
My question is about:
std::round(value * precision * 1000.0) != std::round(value * precision) * 1000.0

Reply:
The way it was rewritten (unless I'm missing something), it should be as follows since the value is scaled by `scale_` when calling `roundData` instead of inside of `roundData`:
```
std::round(value * precision * 1000.0) == std::round((value * 1000.0) * precision)
```